lib/tools.js

```diff
@@ -610,14 +610,25 @@ function scanDirectory(dirName, list, regExp) {
         }
     }
 }
-// Get list of all installed adapters and controller version on this host
+
+/**
+ * Get list of all installed adapters and controller version on this host
+ * @param {string} hostRunningVersion version of the running js-controller
+ * @returns {object} object containing information about installed host
+ */
 function getInstalledInfo(hostRunningVersion) {
     const result = {};
-    const path = __dirname + '/../';
+    const path = `${__dirname}/../`;
     // Get info about host
-    const ioPackage = JSON.parse(fs.readFileSync(path + 'io-package.json', 'utf8'));
-    const package_  = fs.existsSync(path + 'package.json') ? JSON.parse(fs.readFileSync(path + 'package.json', 'utf8')) : {};
-    const regExp    = new RegExp('^' + module.exports.appName + '\\.', 'i');
+    let ioPackage;
+    try {
+        ioPackage = JSON.parse(fs.readFileSync(`${path}io-package.json`, 'utf8'));
+    } catch (e) {
+        console.error(`Cannot get installed host information: ${e}`);
+        return result;
```

Comment:
really return not just not  add this result entry? because the other scanDirectory addstuff too .... so only log an error?